crates/fluvio-version-manager/src/command/install.rs

```diff
@@ -0,0 +1,285 @@
+//! Install Command
+//!
+//! Downloads and stores the sepecific Fluvio Version binaries in the local
+//! FVM cache.
+
+use std::fs::{File, create_dir, copy, write};
+use std::io::{Cursor, Read};
+use std::path::PathBuf;
+
+use anyhow::{Error, Result};
+use clap::Parser;
+use colored::Colorize;
+use serde::{Deserialize, Serialize};
+use semver::Version;
+use sha2::{Digest, Sha256};
+use surf::StatusCode;
+use tempfile::TempDir;
+use url::Url;
+
+use fluvio_hub_util::HUB_REMOTE;
+use fluvio_hub_util::fvm::{Client, PackageSet, DEFAULT_PKGSET, Channel};
+
+use crate::GlobalOptions;
+use crate::common::TARGET;
+use crate::common::fluvio::{fluvio_bin_path, create_fluvio_dir};
+use crate::common::notify::Notify;
+use crate::common::workdir::fvm_versions_path;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct VersionManifest {
+    pub channel: Channel,
+    pub version: Version,
+}
+
+/// The name printed to CLI in place of `default` for the Package Set
+const DEFAULT_PACKAGE_SET_DISPLAY_NAME: &str = "fluvio";
+
+/// The name of the manifest file for the Package Set
+const PACKAGE_SET_MANIFEST_FILENAME: &str = "manifest.json";
+
+/// The `install` command is responsible of installing the desired Package Set
+#[derive(Debug, Parser)]
+pub struct InstallOpt {
+    #[command(flatten)]
+    global_opts: GlobalOptions,
+    /// Version to install
+    #[arg(index = 1, default_value_t = Channel::Stable)]
+    version: Channel,
+    /// Registry used to fetch Fluvio Versions
+    #[arg(long, env = "HUB_REGISTRY_URL", default_value = HUB_REMOTE)]
+    registry: Url,
+}
+
+impl InstallOpt {
+    pub async fn process(&self) -> Result<()> {
+        if !fluvio_bin_path()?.exists() {
+            create_fluvio_dir()?;
+        }
+
+        self.install_package().await?;
+        Ok(())
+    }
+
+    async fn install_package(&self) -> Result<()> {
+        let pkgset = self.fetch_pkgset().await?;
+        let version_path = fvm_versions_path()?.join(self.version.to_string());
+
+        if version_path.exists() {
+            self.notify_warn(format!(
```

Comment:
Actually yes! I did like this before to check on SemVer in case a channel was provided, so I can tell if an update is available. But now that we will have `fvm update` its not required anymore!

Reply:
Now we fetch after checking 71bfaac